libs/mock-block-dock/src/debug-view/dev-tools/datastore-graph-visualization.tsx

```diff
@@ -179,7 +179,7 @@ const getSubgraphEdgesAsEChartEdges = (
     return typedEntries(inner).flatMap(([revisionId, outwardEdges]) => {
       return outwardEdges.flatMap((outwardEdge) => {
         const sourceRevisions = typedKeys(
-          subgraph.vertices[sourceBaseId]!,
```

Comment:
Did you encounter an issue with this? 🤔 (I shouldn't have used `!` there anyway but the `mustBeDefined` helper instead)

I don't understand how we'd have a `sourceBaseId` without it being in the `vertices`

Reply:
Yes it broke locally for me without. I think it broke specifically when trying to walk one of the `CONSTRAINED_ON` edges that weren't in the subgraph (the number data type specifically).

Reply:
Yes, we are doing that right now. The functions for adding the different ontology types to the graph (`addEntityTypesToSubgraphByMutation` for example) don't check if the types exist beforehand. What is the desired behavior if the vertices don't exist?

Reply:
Let's jump on a call

Reply:
Discussed offline: we want to change the logic of the `buildSubgraph` helpers to make use of a "two step" process, first adding vertices then resolving those vertex edges